apigee_edge.module

```diff
@@ -1270,6 +1270,36 @@ function template_preprocess_app_credential(array &$variables) {
   }
 }
 
+/**
+ * Forcing user email address to lowercase for ApigeeX/Hybrid Org.
+ * @see https://github.com/apigee/apigee-edge-drupal/issues/730
+ */
+function apigee_edge_user_email_check(&$account) {
+  $email = $account->getEmail();
+  try {
+    /** @var \Drupal\apigee_edge\SDKConnectorInterface $sdk_connector */
+    $sdk_connector = \Drupal::service('apigee_edge.sdk_connector');
+    $org_controller = \Drupal::service('apigee_edge.controller.organization');
+    /* @var \Apigee\Edge\Api\Management\Entity\Organization $organization */
+    $organization = $org_controller->load($sdk_connector->getOrganization());
+
+    // Check if org is ApigeeX.
+    if ($organization && ('CLOUD' === $organization->getRuntimeType() || 'HYBRID' === $organization->getRuntimeType())) {
+       // Apigee X Org is only accepting lower case email address,
+       // therefore preventing the form from creating account
+       // with uppercase email address.
+       if (preg_match('/[A-Z]/', $email)) {
+        $account->set('mail', strtolower($email));
```

Comment:
Thank you @mxr576 for reviewing this PR.

Do you mean to say, that instead of forcefully converting the email address to lowercase we should put a constraint on the email field so the user gets an error message while user create/update, if capitalized email address found for Apigee-X Org.

Reply:
Yes, exactly - And developer sync is also going to fail (it already works like this) when entity validation fails.